operations/overlay_compute_ngd.yml

```diff
@@ -0,0 +1,36 @@
+id: overlay_compute_ngd
+subclass_of: overlay
+name: overlay compute ngd
+description: This operation computes the normalized Google distance (co-occurrence frequency) in PubMed abstracts and adds virual edges between qnodes AND/OR knodes AND/OR results edge bindings.
+examples: 
+- input: overlay/messages/07_input_ngd.json
+  output: overlay/messages/08_output_ngd.json
+output_guarantees:
+- At least 1 new edge between qnodes
+allowed_changes:
+- add kedges
+- add qedges
+- add bindings to results
+parameters: 
+  type: object
+  properties: 
+    default_value:
+      description: The default value of the normalized Google distance (if its value cannot be determined)
+      type: string
```

Comment:
This is awkward. The implication is that the string will be converted to a number internally?  I want it to be `type: number`, but that does not admit infinity (this is a limitation of JSON). For instances of type string, `minimum` and `maximum` will not do anything. I think what I'd do is switch to `type: number`, remove `maximum` since it's redundant, and remove `default` too. Instead, the server could use -1 to indicate NaN, or just leave out the edge.

Reply:
Hmmm... @dkoslicki any thoughts on setting this to `type: number` and hen handling a `default_value` of -1 as infinity?

Reply:
@finnagin no objections to handling -1 as infinity, we just need to be absolutely sure that's the case, otherwise those will be the best edges!

Reply:
why not an absurdly large number? like 999
Also, should be Inf instead of NaN?

Reply:
@finnagin mind clarifying the use of NaN? The denominator of NGD should always be non-zero (in non-pathologic cases), so I think we were using that when we failed to find any publications for any entities (or their co-occurrence). In that case, we should treat them as 0, so NaN's shouldn't really show up. I'd really like `np.inf` as the default value, but I guess `N` would work where `N` is the total number of PubMed concepts.

Reply:
@dkoslicki the [default value is actually currently using inf](https://github.com/NCATSTranslator/OperationsAndWorkflows/blob/83ee1cd3c6b28d025f4e9165c3bb0ae9de831407/operations/overlay_compute_ngd.yml#L22-L23) not NaN. 

Since `default_value` is just the value returned when we can't find publications does it make sense to have this as a parameter in the first place? Shouldn't it just always be infinity?

Reply:
It was I who subtly suggested NaN over inf. The description says "its value cannot be determined", not that we have determined the value to be very large. If we just mean to fill in this value when we find 0 publications, then it should indeed be very large, and the name and/or description of this field should change.

Reply:
This has now been changed to be a number with -1 treated as infinity. 

Though again, I kinda feel like we could just remove the `default_value` parameter altogether and always treat it as infinity. At least I can't think of a single case we would want to use anything but infinity when no publications are found. Does anyone else have any thoughts?

Reply:
I agree, seems like an unneeded parameter

Reply:
In that case I'll go ahead and remove it.

Reply:
I've now removed the `default_value` parameter and added a note to the description mentioning that infinity will be returned when no publications are found.